frontend/src/lib/Buttons.ts

```diff
@@ -18,112 +18,306 @@ import AddIcon from '@material-ui/icons/Add';
-    action,
-    icon: AddIcon,
-    id: 'createNewRunBtn',
-    outlined: true,
-    primary: true,
-    title: 'Create run',
-    tooltip: 'Create a new run within this pipeline',
-  }),
-  refresh: action => ({
-    action,
-    id: 'refreshBtn',
-    title: 'Refresh',
-    tooltip: 'Refresh the list',
-  }),
-  restore: action => ({
-    action,
-    disabled: true,
-    disabledTitle: 'Select at least one resource to restore',
-    id: 'archiveBtn',
-    title: 'Archive',
-    tooltip: 'Archive',
-  }),
-  upload: action => ({
-    action,
-    icon: AddIcon,
-    id: 'uploadBtn',
-    outlined: true,
-    title: 'Upload pipeline',
-    tooltip: 'Upload pipeline',
-  }),
-};
-
-export default Buttons;
+export default class Buttons {
+  private _props: PageProps;
+  private _refresh: () => void;
+  private _urlParser: URLParser;
+
+  constructor(pageProps: PageProps, refresh: () => void) {
+    this._props = pageProps;
+    this._refresh = refresh;
+    this._urlParser = new URLParser(pageProps);
+  }
+
+  public cloneRun(getSelectedIds: () => string[], useCurrent: boolean): ToolbarActionConfig {
+    return {
+      action: () => this._cloneRun(getSelectedIds()),
+      disabled: !useCurrent,
+      disabledTitle: useCurrent ? undefined : 'Select a run to clone',
+      id: 'cloneBtn',
+      title: 'Clone run',
+      tooltip: 'Create a copy from this run\s initial state',
+    };
+  }
+
+  public collapseSections(action: () => void): ToolbarActionConfig {
+    return {
+      action,
+      icon: CollapseIcon,
+      id: 'collapseBtn',
+      title: 'Collapse all',
+      tooltip: 'Collapse all sections',
+    };
+  }
+
+  public compareRuns(getSelectedIds: () => string[]): ToolbarActionConfig {
+    return {
+      action: () => this._compareRuns(getSelectedIds()),
+      disabled: true,
+      disabledTitle: 'Select multiple runs to compare',
+      id: 'compareBtn',
+      title: 'Compare runs',
+      tooltip: 'Compare up to 10 selected runs',
+    };
+  }
+
+  public delete(getSelectedIds: () => string[], resourceName: 'pipeline' | 'recurring run config',
+    callback: (selectedIds: string[], success: boolean) => void, useCurrent: boolean): ToolbarActionConfig {
+    return {
+      action: () => resourceName === 'pipeline' ?
+        this._deletePipeline(getSelectedIds(), callback, useCurrent) :
+        this._deleteRecurringRun(getSelectedIds()[0], useCurrent, callback),
+      disabled: !useCurrent,
+      disabledTitle: useCurrent ? undefined : `Select at least one ${resourceName} to delete`,
+      id: 'deleteBtn',
+      title: 'Delete',
+      tooltip: 'Delete',
+    };
+  }
+
+  public disableRecurringRun(getId: () => string): ToolbarActionConfig {
+    return {
+      action: () => this._setRecurringRunEnabledState(getId(), false),
+      disabled: true,
+      disabledTitle: 'Run schedule already disabled',
+      id: 'disableBtn',
+      title: 'Disable',
+      tooltip: 'Disable the run\'s trigger',
```

Comment:
nit: maybe "Prevent[/Stop] this from scheduling[/creating] additional runs"

Reply:
@ajayalfred what do you think of this suggestion? I also think it can use a little clearing.